src/PPN-microbench/memory.cpp

```diff
@@ -0,0 +1,81 @@
+#include <PPN-microbench/memory.hpp>
+
+
+// Function to generate a random index within a given threshold
+inline u64 random_index(u64 threshold) { return rand() % threshold; }
+
+// Function to measure cache/memory latency using pointer chasing
+double measure_latency(u64 size, u64 nbIterations) {
+    std::vector<void *> memblock(size);
+
+    // Initialize the memory block with pointers to itself
+    for (u64 i = 0; i < size; ++i) {
+        memblock[i] = &memblock[i];
+    }
+
+    // Shuffle the pointers to create a random access pattern
+    for (u64 i = size - 1; i > 0; --i) {
+        u64 j = random_index(i + 1);
+        std::swap(memblock[i], memblock[j]);
+    }
+
+    double total_latency = 0.0;
+
+    // Perform the latency measurement 10 times for better accuracy
+    for (int run = 0; run < 10; ++run) {
+        void *p = memblock[0];
+        auto start = std::chrono::high_resolution_clock::now();
+
+        // Pointer chasing loop
+        for (u64 i = 0; i < nbIterations; ++i) {
+            p = *(void **)p;
+        }
+
+        auto end = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double, std::nano> elapsed = end - start;
+
+        // Accumulate the latency
+        total_latency += elapsed.count() / nbIterations;
+    }
+
+    // Calculate the average latency over 10 runs
+    return total_latency / 10.0;
+}
+
+// Constructor
+Memory::Memory(std::string name, int nbIterations)
```

Comment:
Since your benchmark will always have the same name and it will always have the same number of iterations, you can declare the parameters in raw form in the constructor of the parent class without having to indicate them in the child class :

```c++
Memory::Memory() : Microbench("Memory", 1) {
```

By unsing that, you're sure that your benchmark is always the same and you don't to have to declare all variables in `conductor_main.cpp` if they are not specific to your child class